### PR TITLE
Adds Plushies +Arcade Changes

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Adds in plushies; all sprites by @FullofSkittles (thank you!)

Plushies can be hugged or snugged---that said, they're mostly just decorative/for fun.

Also adds in alternative carp plushies

The reason plushise and carp plushies are separate is I wanted to keep some of the unique things carp plushies do intact, while also not messing with the dehydrated carp plushie traitor item.

pics:

floofs

Arcade related

- Instead of having a ton of different mechs all individually weighted, it's just a random item generator that picks from the different kind of mech toys
- Carp plushies will now be randomly colored instead of always purple/pink.
- Adds in random plushies

### CODE REVIEW

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1095,6 +1103,165 @@ obj/item/toy/cards/deck/syndicate/black
 	playsound(src.loc, bitesound, 20, 1)
 	return ..()
 
+
+/obj/random/carp_plushie
+	name = "Random Carp Plushie"
+	desc = "This is a random plushie"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "carpplushie"
+	item_to_spawn()
```
Comment (tigercat2000) [inline code/game/objects/items/toys.dm]: why is this relative-pathed

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -522,6 +522,14 @@
 			return
 	..()
 
+/obj/random/prize
+	name = "Random Mech Prize"
+	desc = "This is a random prize"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "ripleytoy"
+	item_to_spawn()
```
Comment (tigercat2000) [inline code/game/objects/items/toys.dm]: same here

**Comment by TheDZD** [comment]
Err, seems kind of odd that there is a carpplushie type, rather than just having carp be a subtype of plushie.

**Comment by Fox-McCloud** [comment]
I explained that; the behavior of carp plushies is different from that of regular plushies, not to mention the special behaviors+interactions of the traitor item that is a carp plushie (yes, it's a subtype of the carp plushie).

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -1095,6 +1104,167 @@ obj/item/toy/cards/deck/syndicate/black
...
 
+
+/obj/random/carp_plushie
+	name = "Random Carp Plushie"
+	desc = "This is a random plushie"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "carpplushie"
+
+/obj/random/carp_plushie/item_to_spawn()
+	return pick(typesof(/obj/item/toy/carpplushie)) //can pick any carp plushie, even the original.
```
Comment (ZomgPonies) [inline code/game/objects/items/toys.dm]: subtypesof

Comment (TheDZD) [inline code/game/objects/items/toys.dm]: It seems like you do want to include the base type here as well.

Comment (Fox-McCloud) [inline code/game/objects/items/toys.dm]: Yeah, this needs to explicitly be types off; it's why I left the comment.

**Comment by Mijhead** [comment]
no nymph plushies ;-;

**Comment by Plotron** [comment]
nymphs are gross

**Comment by Spacemanspark** [comment]
Plotron is gross.

**Comment by Tastyfish** [comment]
Now I want a parrot plushie that spouts catchphrases when squeezed.

**Comment by TheDZD** [comment]
Cooooonflicts.

File: code/game/objects/items/toys.dm [hunk 4]
```diff
@@ -1095,6 +1104,167 @@ obj/item/toy/cards/deck/syndicate/black
...
+	attack_verb = list("poofed", "bopped", "whapped","cuddled","fluffed")
+
+/obj/item/toy/plushie/attack(mob/M as mob, mob/user as mob)
+	playsound(loc, poof_sound, 20, 1)	// Play the whoosh sound in local area
+	return ..()
+
+/obj/item/toy/plushie/attack_self(mob/user as mob)
+	var/cuddle_verb = pick("hugs","cuddles","snugs")
+	user.visible_message("<span class='notice'>[user] [cuddle_verb] the [src].</span>")
+	playsound(get_turf(src), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
```
Comment (Tastyfish) [inline code/game/objects/items/toys.dm]: Why doesn't this use poof_sound?

**Comment by Fox-McCloud** [comment]
@TheDZD 

Should be ready, once Travis passes.

**Pull request merged**